components/script/dom/websocket.rs

```diff
@@ -333,6 +333,31 @@ impl WebSocketMethods for WebSocket {
 
     // https://html.spec.whatwg.org/multipage/#dom-websocket-send
     fn Send(&self, data: USVString) -> Fallible<()> {
+
+        let bufferedAmount = self.buffered_amount.get() + (data.0.as_bytes().len() as u32)
+
+        self.Send_Impl(bufferedAmount)
+
+        let _ = my_sender.lock().unwrap().send_message(Message::Text(data.0));
+
+        Ok(())
+    }
+
+    // https://html.spec.whatwg.org/multipage/#dom-websocket-send
+    fn Send(&self, data: Blob) -> Fallible<()> {
+
+        let bufferedAmount = self.buffered_amount.get() + (data.size as u32)
+
+        self.Send_Impl(bufferedAmount)
+
+        let _ = data.send(my_sender.lock().unwrap().send_message)
```

Comment:
@dzbarsky I wasn't and I'm still not sure how to use the send function for Blob. Looking at the API my best guess was it needs to be sent a function that it then calls https://github.com/servo/servo/blob/master/components/script/dom/blob.rs#L73. How should I be using it instead?